server/src/shared/query/pagination.ts

```diff
@@ -0,0 +1,50 @@
+import { IDeferredJoinParams, IKeysetParams } from 'shared/types/pagination';
+import db from 'shared/database';
+
+export async function deferredJoinPagination({
+  tableName,
+  indexedColumnName,
+  selectAttributes,
+  limit,
+  orderByDirection,
+  pageNumber,
+}: IDeferredJoinParams) {
+  const offset = (pageNumber - 1) * limit;
+  const innerQueryTableName = `${indexedColumnName}s_page`;
+  const attributes = selectAttributes || '*';
+  return db
```

Comment:
Could you please write out somewhere a short description of what is happening in this method?

Reply:
An explanation is provided in separate comment. ⬇️